rest/data/system_metrics_test.go

```diff
@@ -105,41 +106,65 @@ func (s *systemMetricsConnectorSuite) TearDownSuite() {
 
 func (s *systemMetricsConnectorSuite) TestFindSystemMetricsByTypeFound() {
 	// no execution
-	opts := dbModel.SystemMetricsFindOptions{
+	findOpts := dbModel.SystemMetricsFindOptions{
 		TaskID:         "task1",
 		EmptyExecution: true,
 	}
-	data, err := s.sc.FindSystemMetricsByType(s.ctx, "uptime", opts)
+	downloadOpts := dbModel.SystemMetricsDownloadOptions{MetricType: "uptime"}
+	data, nextIdx, err := s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
 	s.Require().NoError(err)
-	s.Equal("execution 1", string(data))
+	s.Equal("execution 1\nchunk2", string(data))
+	s.Equal(2, nextIdx)
 
 	// execution specified
-	opts.Execution = 0
-	opts.EmptyExecution = false
-	data, err = s.sc.FindSystemMetricsByType(s.ctx, "uptime", opts)
+	findOpts.Execution = 0
+	findOpts.EmptyExecution = false
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
 	s.Require().NoError(err)
-	s.Equal("execution 0", string(data))
+	s.Equal("execution 0\nchunk2", string(data))
+	s.Equal(2, nextIdx)
+
+	// paginated
+	downloadOpts.PageSize = 5
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
+	s.Require().NoError(err)
+	s.Equal("execution 0\n", string(data))
+	s.Equal(1, nextIdx)
+
+	downloadOpts.StartIndex = 1
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
+	s.Require().NoError(err)
+	s.Equal("chunk2", string(data))
+	s.Equal(2, nextIdx)
+
+	downloadOpts.StartIndex = 2
+	data, nextIdx, err = s.sc.FindSystemMetricsByType(s.ctx, findOpts, downloadOpts)
+	s.Require().NoError(err)
+	s.Nil(data)
+	s.Equal(2, nextIdx)
```

Comment:
Should this also test the case where the page size is large enough to read multiple chunks (but not all of them at once)?

Reply:
That is tested in the model package, I just want to test that it is calling the right thing.